### ISSUE TITLE
Editor: User can't upload media to Jetpack site, receives null API error response 

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-iOS#10949]
### Expected behavior

I expect to be able to upload media to a Jetpack site from within the app editor. If the upload fails, I expect a clear error message about the failure.

### Actual behavior

A user is unable to upload media from the app to a Jetpack site, although there are no apparent problems with the Jetpack connection. Logs from the app show the following errors, including a null error from the wp.com REST API:

```
2019-01-09 12:04:36:754 Unable to download image for attachment with url = placeholder://. Details: Optional("unsupported URL")
2019-01-09 12:04:36:755 🔵 Tracked: media_service_upload_started, properties: {
"blog_id" = 155280537;
}
2019-01-09 12:04:36:785 Unable to download image for attachment with url = placeholder://. Details: Optional("unsupported URL")
2019-01-09 12:04:47:799 🔵 Tracked: media_service_upload_failed, properties: {
"error_code" = 7;
"error_description" = "Error Domain=WordPressKit.WordPressComRestApiError Code=7 \"(null)\"";
"error_domain" = "WordPressKit.WordPressComRestApiError";
```

### Steps to reproduce the behavior

The issue is consistently reproduced on the user's site, but we haven't been able to reproduce it on our end. (Internal ref: p4a5px-2ky-p2 / 1694516-zen )

Reported using WordPress iOS 11.4.1 on iOS 12.1.2

**Comment by hartington** [comment]
Just wondering if anyone could shed any light on this, it’s really annoying I’m unable to upload photos from the iPhone app. The original ticket I opened has now been closed down too.

**Comment by kristarella** [comment]
@hartington I believe the issue reported is for the editor specifically, do you have the same problem uploading to the media library and then inserting into the post?
Does this happen for any amount of images? Single and multiple images?
Have you tried deleting and reinstalling the WordPress app?

**Comment by kristarella** [comment]
@hartington 

I believe the issue reported is for the editor specifically, do you have the same problem uploading to the media library and then inserting into the post?
Does this happen for any amount of images? Single and multiple images?
Have you tried deleting and reinstalling the WordPress app?

**Comment by supernovia** [comment]
Noting we've got another user experiencing this too. #1904564-t

They are able to use wordpress.com/media to post to one site but not another. Same in their app, authenticating with just wp-admin usernames and passwords.

**Comment by hartington** [comment]
Hi,

Apologies on the delay there was a potential security risk with the WordPress app, so they disconnected my sites from the app, I’ve had problems getting them to log back in, however I’ve managed this evening.

Yes I have the same problem uploading to the media library, exactly the same thing happens.

It happens for any amount of images yes, whether it’s one or three. The same issue.

I have tried deleting and re-installing the app yes.

Hope you’re able to help.

Best wishes,

Pete.

**Comment by designsimply** [comment]
Thank you for the update @hartington! I wanted to mention for others following along that we are also tracking your case via email (internal reference: 1694516-zen).

For the larger issue, I came here to note there was an additional report in 1879361-zen with this explanation given:

<blockquote>I am having some trouble uploading images to my site from my iPhone. It’s an iPhoneXr with what I believe is the latest WordPress app and latest IOS. I keep getting this error whether I try to upload by embedding or upload a featured image in the post settings. These images are screenshots from my phone, but that shouldn’t be affecting things should it?</blockquote>

`The operation couldn’t be completed. (WordPressKit.WordPressComRestApiError error 7.)`

<img src="https://user-images.githubusercontent.com/1119271/55589745-4d43b980-56ee-11e9-9225-2dffa56290e4.PNG" width="270" alt="IMG_0837">

**Comment by designsimply** [comment]
Updating to note that 1904564-zen identified the `WP-Super-Cache` plugin was the source of the problem and that checking the plugins settings and reporting the issue to the plugin developer directly are the recommended next steps for that case.

**Comment by designsimply** [comment]
Noting that the explanation for the following log is that Aztec by default will add an image placeholder when a new image is inserted, then will try to download it from the provided url ( placeholder:// in this case) and it will fail. But this parallel to the upload process of the inserted image from the device. It is not a bug. 
<sup>(internal reference: p4a5px-2mM-p2#comment-9630)</sup>

```
Unable to download image for attachment with url = placeholder://. Details: Optional("unsupported URL")
```

For this case, the important log to look for is this one:

```
2019-01-09 12:04:47:799 🔵 Tracked: media_service_upload_failed, properties: {
"error_code" = 7;
"error_description" = "Error Domain=WordPressKit.WordPressComRestApiError Code=7 \"(null)\"";
"error_domain" = "WordPressKit.WordPressComRestApiError";
```

**Comment by mediajolt001** [comment]
I don’t think this issue has been resolved. I’m getting this error when trying to upload an image on any of my iOS devices. I don’t have the Wp-Super-Cache plugin (and never have).

**Comment by mediajolt001** [comment]
> @hartington
> 
> I believe the issue reported is for the editor specifically, do you have the same problem uploading to the media library and then inserting into the post?
> Does this happen for any amount of images? Single and multiple images?
> Have you tried deleting and reinstalling the WordPress app?

Yes. It also happens using the web dashboard.

**Comment by designsimply** [comment]
> It also happens using the web dashboard.

@mediajolt001 if you are unable to upload media using a web browser, that indicates your case may not be related to the mobile app and could be a problem with your site setup specifically. You will want to rule any problems specific to your setup as a first step in your case. This is not the right place to troubleshoot. May I please ask that you contact Jetpack support at https://jetpack.com/contact-support/?rel=support and include a short list of steps to reproduce the problem on the web, your site address, and the exact wording of the error you see when the problem happens. If you are not using Jetpack or don't know what that is, let me know a bit more about your site setup and I will try to direct you better.
